Libraries/Components/DatePicker/DatePickerIOS.ios.js

```diff
@@ -118,22 +128,32 @@ var DatePickerIOS = React.createClass({
 
   render: function() {
     var props = this.props;
+
+    if (props.date && typeof props.date !== 'number') {
+      props.date = props.date.getTime();
+    }
+    if (props.minimumDate && typeof props.minimumDate !== 'number') {
+      props.minimumDate = props.minimumDate.getTime();
+    }
+    if (props.maximumDate && typeof props.maximumDate !== 'number') {
+      props.maximumDate = props.maximumDate.getTime();
+    }
+
     return (
       <View style={props.style}>
         <RCTDatePickerIOS
           ref={ picker => this._picker = picker }
           style={styles.datePickerIOS}
-          date={props.date.getTime()}
-          maximumDate={
-            props.maximumDate ? props.maximumDate.getTime() : undefined
-          }
-          minimumDate={
-            props.minimumDate ? props.minimumDate.getTime() : undefined
-          }
+          date={props.date}
+          maximumDate={props.maximumDate}
+          minimumDate={props.minimumDate}
           mode={props.mode}
           minuteInterval={props.minuteInterval}
           timeZoneOffsetInMinutes={props.timeZoneOffsetInMinutes}
           onChange={this._onChange}
+          onDateChange={() => {
+            //Do nothing, just remove warn of unset function
+          }}
```

Comment:
I think it's can be done in another way, by changing the name of the native function from `onChange` to `onDateChange`, this warning is triggered because the js props are apply to native props, so that's require a function named `onDateChange`
@christopherdro, @ide: What do you think ?